osbuild/util/osrelease.py

```diff
@@ -33,7 +33,7 @@ def parse_files(*paths):
                 if line[0] == "#":
                     continue
                 key, value = line.split("=", 1)
-                osrelease[key] = value.strip('"')
+                osrelease[key] = value.strip(" \n\t'\"")
```

Comment:
I wonder how a `\n` could end up in there, it seems we already split on line but maybe I'm missing something?

Reply:
This is for CoreOS, specifically for ostree (OSTREE_VERSION). I'm not sure why single quotes are being used here, I'll try to see if we can change  it on our end. However, I'm not sure if this might impact other things around, so we may need to keep this change in place temporarily.

```
NAME="Fedora Linux"
VERSION="40.20241106.dev.0 (CoreOS)"
ID=fedora
VERSION_ID=40
VERSION_CODENAME=""
PLATFORM_ID="platform:f40"
PRETTY_NAME="Fedora CoreOS 40.20241106.dev.0"
ANSI_COLOR="0;38;2;60;110;180"
LOGO=fedora-logo-icon
CPE_NAME="cpe:/o:fedoraproject:fedora:40"
HOME_URL="https://getfedora.org/coreos/"
DOCUMENTATION_URL="https://docs.fedoraproject.org/en-US/fedora-coreos/"
SUPPORT_URL="https://github.com/coreos/fedora-coreos-tracker/"
BUG_REPORT_URL="https://github.com/coreos/fedora-coreos-tracker/"
REDHAT_BUGZILLA_PRODUCT="Fedora"
REDHAT_BUGZILLA_PRODUCT_VERSION=40
REDHAT_SUPPORT_PRODUCT="Fedora"
REDHAT_SUPPORT_PRODUCT_VERSION=40
SUPPORT_END=2025-05-13
VARIANT="CoreOS"
VARIANT_ID=coreos
OSTREE_VERSION='40.20241106.dev.0'
```

Reply:
Thank you! This is really helpful, maybe we can just extend for the `'` char then? If you don't mind I would also push a tiny test for this just to make sure that we do not regress here but I'm happy to do that in a followup (or push here if that is okay with you).

Reply:
Thanks, @mvo5! We already have [this test](https://github.com/osbuild/osbuild/blob/main/test/mod/test_util_osrelease.py#L22) that calls parse_files. Maybe only adding the new os-release file is enough?